### PR TITLE
Issue #100: Adding a node hide option to use for placing in blocks or…

### PR DESCRIPTION
**Pull request by serundeputy**
… views.

### CODE REVIEW

File: core/modules/node/node.install [hunk 1]
```diff
@@ -1119,6 +1119,17 @@ function node_update_1013() {
   cache_clear_all();
 }
 
+/**
```
Comment (serundeputy) [inline core/modules/node/node.install]: @quicksketch does this update hook belong inside or outside of the `"addtogroup updates-7.x-to-1.x"`?

**Comment by herbdool** [comment]
Nice @serundeputy! Did you choose to put it in core because there doesn't seem to be hook to accomplish this fully?

A second iteration could also include access denied and redirect.

**Comment by serundeputy** [comment]
@herbdool primary motivator is this will give a system to translate blocks.

> A second iteration could also include access denied and redirect.

sounds good.

**Review by opi — COMMENTED** [review]
What about using "pageless" instead of "node_hider" ? It makes more sense to me...

**Comment by herbdool** [comment]
I second "pageless". Makes more intuitive sense for the 80% of users. I think we want to avoid "node" as much as possible.

**Comment by serundeputy** [comment]
`pageless` it is! Nice suggestion.

**Comment by serundeputy** [comment]
updated to `pageless` and eradicated front end references to the word `node`.

**Comment by olafgrabienski** [comment]
Found another `node` in the settings help text:

> By default nodes will have URLs and be visible as a web page, (...)

Maybe better:

`By default content will have a URL and be visible as a web page`

or:

`By default content items will have URLs and be visible as a web page`

**Comment by olafgrabienski** [comment]
@serundeputy Thanks for updating the PR! I think we need also a permission to view pageless content (or to *bypass* an active pageless setting), so that administrators or content editors are able to view content after adding or editing it.

**Comment by olafgrabienski** [comment]
Some suggestions regarding the user interface:

(1) The **vertical tab title** is `Pageless` at the moment. In my opinion `Pageless settings` fits better to all the other tabs (`Menu settings` and so on).

---

(2) **Streamline settings and help texts**

Currently:

> Would you like to hide these pages from the world?
> 
> - [x] No. Business as usual these pages will be displayed.
> - [ ] Yes. Remove page display to use as content elsewhere on the site.
> 
> *By default nodes will have URLs and be visible as a web page, but you can disable that by selecting 'Yes' here.*
> 
> **Why?**
> 
> Here are some use cases one might choose to hide the page display:
> 
> - Views
>   - One might want to create a view of content that will never show the individual page just the view (slider).
> - Content Blocks
>   - One might want to make a block that has content that needs to be translatable.

Here's a shorter suggestion which also avoids the "disable to enable" situation:

> Show pages as usual.
> 
> - [x] Yes.
> - [ ] No, hide the page display from visitors.
> 
> Pageless content can be displayed elsewhere on the site, for instance in a view or in a block.

**Comment by serundeputy** [comment]
I've implemented @olafgrabienski 's UI suggestions. Thanks @olafgrabienski !

**Comment by serundeputy** [comment]
> I think we need also a permission to view pageless content (or to bypass an active pageless setting), so that administrators or content editors are able to view content after adding or editing it.

@olafgrabienski I've implemented a `bypass pageless setting` permission.

**Comment by olafgrabienski** [comment]
@serundeputy Thanks for the implementations, great!

Re the `bypass pageless setting` permission, had a look at the sandbox site: When the permission is enabled for a certain role, a respective user can visit pageless content, perfect! (I had the impression that `user/1` bypasses the pageless setting anyway now, was that intended?)

Apart from that, in the DBlog I saw several `Cannot modify header information - headers already sent` warnings. Don't know if they're related to the PR.

**Comment by quicksketch** [comment]
backdrop-issues#100

**Comment by quicksketch** [comment]
backdrop/backdrop-issues#100

**Comment by quicksketch** [comment]
Issue #100

File: core/modules/node/node.entity.inc [hunk 2]
```diff
@@ -62,6 +62,13 @@ class Node extends Entity {
    */
   public $status;
 
+  /**
+   * The setting to hide node display.
+   *
+   * @var string
+   */
+  public $node_hider;
```
Comment (opi) [inline core/modules/node/node.entity.inc:70]: `$pageless` instead ? Maybe comment should be updated too.

**Comment by herbdool** [comment]
I think this should be just a checkbox to follow more closely other content type options. And we'll need provide some more explanation so people know about possible uses for this. And that there's a permission to bypass the restriction.

**Comment by olafgrabienski** [comment]
@herbdool I've thought about what to say in the "Pageless settings" tab.

**(a) Current status**

Pageless settings
Showing pages (vs. Hiding pages)

Show pages as usual.
[x] Yes.
[ ] No, hide the page display from visitors.

Pageless content can be displayed elsewhere on the site, for instance in a view or in a block.

**(b) New proposal**

Pageless settings
Showing page display (vs. Hiding page display)

[ ] Pageless

Pageless content is not accessible on its own page, except for administrators or others with a bypass permission. The content can be displayed without restrictions elsewhere on the site, for instance in a slideshow view or in an existing content block.

---

I'm not a native English speaker, so feel free to correct the language. Apart from that, I find the pageless concept hard to explain in a few words, open for other suggestions. The help text however shouldn't be much more verbose at this place, in my opinion.

Btw, another idea: What if we put the "Pageless" option on the "Display settings" tab?

**Comment by herbdool** [comment]
Just getting back to this. Here is what I propose:

> Tab title: Full Page settings
> 
> [ ] Hide full page display for non-administrators
> 
> The content can still be inserted into a slideshow view or as a block and be visible, but will be hidden when accessed as a full page.

Maybe this option could even be put under the tab "Display Settings" and avoid the awkward tab title.

**Comment by olafgrabienski** [comment]
@herbdool I definitely like the idea to put the option under the "Display settings" tab.

The shorter help text is also much better. In my opinion it makes however more sense
- to start the phrase with the direct result of the setting (hides full page content),
- to mention visibility in a slideshow or as a block as examples ("for instance ...").

Something like this:

> The content will be hidden when accessed as a full page, but can be made visible otherwise, for instance in a slideshow view or as a block.

@serundeputy What do you think of the last suggestions? And, did you see https://github.com/backdrop/backdrop/pull/2049#pullrequestreview-86838857?

**Comment by herbdool** [comment]
Is this still relevant? I could tweak and reroll if people still want this.

**Comment by olafgrabienski** [comment]
@herbdool I'm still interested (as you can imagine), in my opinion it's still a good feature, nice and simple compared to *Rabbit Hole*.

**Comment by olafgrabienski** [comment]
@herbdool Did you see the message "This branch has conflicts that must be resolved"? Apart from that, the last ideas were to modify the UI text a bit and to put the "Hide full page display" option under the "Display settings tab". I still like these ideas, what do you think about it? And/or do you have new suggestions?

**Comment by herbdool** [comment]
I've provided an alternative PR https://github.com/backdrop/backdrop/pull/2286

**Comment by quicksketch** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/2286.

**Pull request closed without merging**